### PR TITLE
Allow expands for retrieval of comments

### PR DESCRIPTION
**Pull request by matthewkeenan**
Allow expands for retrieval of comments

Fixes: #949

### CODE REVIEW

**Comment by matthewkeenan** [comment]
Travis CI Errors are the not linked with this pull request changes. The exact same failures can be seen earlier still open pull requests.

**Comment by studioj** [comment]
> Travis CI Errors are the not linked with this pull request changes. The exact same failures can be seen earlier still open pull requests.

looks like the maintainers want to have the ci system back up and running before agreeing to any new merges.

if you want to contribute to get yr pr through have a look here => https://github.com/pycontribs/jira/issues/896

**Comment by studioj** [comment]
@matthewkeenan  please apply last changes in the master branch and resolve the conflicts. 
Thank you

File: tests/tests.py
Comment (studioj) [inline tests/tests.py:1108]: seems like this is a cleanup part. if this test fails on the jira cloud instance then it wont be cleaned up.
maybe delete any comments first on issue_1

Comment (matthewkeenan) [reply tests/tests.py:1108]: I presume you mean to move both self.assert calls down after the comment1.delete() and comment2.delete() calls, thus guaranteeing they get removed regardless of whether the asserts fail or not.

Comment (matthewkeenan) [reply tests/tests.py:1106]: Latest commit I've moved comment deletions to be before assert calls. Thanks for the review.

Comment (studioj) [reply tests/tests.py:1106]: Black failed

**Comment by studioj** [comment]
> CI tests are failing on "Lint with tox" stage of CI, yet I cannot see what the actual errors are.

Black failed

**Comment by matthewkeenan** [comment]
CI issues have been fixed and looks to be all passing now.

File: jira/client.py
Comment (studioj) [inline jira/client.py:1598]: are these fields? maybe add that too if these are field names 
maybe provide an example?

Comment (matthewkeenan) [reply jira/client.py]: Just updated the comment's indicating what can be specified, e.g. renderedBody or properties.

**Comment by studioj** [comment]
@matthewkeenan some conflicts to resolve :-)

**Comment by studioj** [comment]
changes seem good to me.
@ssbarnea should give his stamp of approval => maybe we should make this expand possible for more fields?

**Comment by matthewkeenan** [comment]
Just wanted to check if there's anything else that I need to specifically do help get this pull request accepted. Thanks. Matt

File: jira/client.py
Comment (ssbarnea) [inline jira/client.py:1643]: Type is removed here, it should stay in.

File: jira/client.py
Comment (ssbarnea) [inline jira/client.py:1621]: add type hint to expand

**Review by ssbarnea — CHANGES_REQUESTED** [review]

File: jira/client.py
Comment (matthewkeenan) [reply jira/client.py]: I pushed the recommended updates.

File: jira/client.py
Comment (matthewkeenan) [reply jira/client.py]: Recommended update has been pushed.

**Review by matthewkeenan — COMMENTED** [review]
I've commited the requested changes.

**Review by ssbarnea — APPROVED** [review]

**Pull request merged**
